parlai/scripts/generate_model_card.py

```diff
@@ -0,0 +1,1628 @@
+        # using _search_make_li
+        for title in dropdowns:
+            values = self.hyperparams.get(title)
+            if title == 'top_not_dropdown':
+                special = self.hyperparams['maybe_special'][OTHER]
+                keys = self.hyperparams['always_include'][OTHER] + list(special.keys())
+                default_value = 'Not specified'
+                process_func = [None] * (len(keys)) + [extra_func] * len(maybe_keys)
+                process_func += [special[k] for k in keys[-len(special) :]]
+            elif title != 'miscellaneous':
+                common, keys = values
+                if common is None or common == 'None':
+                    commmon_keys = []
+                elif callable(common):
+                    commmon_keys = [key for key in rest_keys if common(key)]
+                else:
+                    commmon_keys = [key for key in rest_keys if common in key]
+                keys.extend(commmon_keys)
+                default_value = None
+                process_func = [None] * len(keys)
+            else:
+                # micellaneous; should be last
+                keys = rest_keys
+                default_value = None
+                process_func = [None] * len(keys)
+
+            L = len(keys)
+            default_values = [default_value] * L
+            before_value = [f'`{key}`: `' for key in keys]
+            after_value = ['`'] * L
+            struct = zip(keys, default_values, before_value, after_value, process_func)
+            content = [self._search_make_li(item, warning=False) for item in struct]
+            content = list(filter(None, content))
+            if len(content) > 0 and title != 'top_not_dropdown':
+                all_content.append(create_dropdown(title, content))
+            else:
+                all_content.insert(0, '\n'.join(content))
+            rest_keys.difference_update(keys)
+
+        return '\n'.join(all_content)
+
+    def user_defined_sec(self, section):
+        if self.model_dict.get(section):
+            return format(self.model_dict[section])
+        if defaults.get(section):
+            return create_warning(defaults.get(section))
+        if self.mode != M_final:
+            msg = f'Missing {section}: Probably need to be grabbed from paper & added by u (the creator)'
+            return create_warning(msg)
+
+    ##########################################
+    # misc/other helper functions
+    ##########################################
+
+    def _check_init_model_keys(self, r, k):
+        return r if self.init_model[k] != self.model_opt[k] else None
+
+    def get_args(self, key_defaults):
+        """
+        Finds the arguments from extra_args_path based on the list of keywords, and
+        replaces or overrides defaults.
+
+        Possible keywords for key_defaults:
+        label_qargs / model_qargs / eval_qargs / section_qargs / user_sections /
+        extra_tasks / extra_models / datasets_used
+        """
```

Comment:
slightly different; instead of reading in the json file every time, using self.all_args to save the info